### PR TITLE
[PLATFORM-403] Profile and Create Stream Page styling/updates

### PR DESCRIPTION
**Pull request by fonty1**
Profile page and Create Stream pages updates, mostly involving styling/responsiveness.

This is mostly css to meet design spec, so to speed things up you guys could let Matt inspect the visual side of things, though please report anything major. 

Some changes to the logic inside the Create Stream Page, the main review item: (StreamPage/Show/ConfigureView/index.jsx)

**Functionality changes:**
- Stream config field items now get appended with an ID so that they can be reordered/live-edited without bugs. IDs get removed before POST/PUTing the stream.
- Stopped duplicate name possibility on live edit
- Ability to hide TOC section/heading via prop 'customStyled'
- Moved historical stuff into the historical component

**Not included:**
- Avatar stuff (designs/backed is a WIP)

### CODE REVIEW

**Comment by juhah** [comment]
I'll just list here all that I come across that seems off 😄 

Stream list:

- Menu disappears (also sort dropdown is hugging the toolbar links):
Screen Shot 2019-03-12 at 15 34 27

Stream detail page:

- Copy stream id button alignment:
Screen Shot 2019-03-12 at 15 27 23
Screen Shot 2019-03-12 at 15 26 50
Screen Shot 2019-03-12 at 15 27 05

- Delete field button gets lost under the side when hovering:
Screen Shot 2019-03-12 at 15 27 47

- API access type goes to next line:
Screen Shot 2019-03-12 at 15 36 46

- API access type dropdown goes to next line (edit):
Screen Shot 2019-03-12 at 15 28 20

- (Very) slight difference in cancel button background colour:
Screen Shot 2019-03-12 at 15 29 09

Profile page:

- Very thin menu in tablet resolution:
Screen Shot 2019-03-12 at 15 33 32

**Comment by fonty1** [comment]
thanks @juhah 

> Menu disappears (also sort dropdown is hugging the toolbar links)

Out of scope for this PR, I will fix this in the Userpages styling branch.

> Copy stream id button alignment (ID prefill)

Thanks, fixed.

> Delete field button gets lost under the side when hovering

Fixed by trimming the tabletish area margins

> API access type goes to next line
> API access type dropdown goes to next line (edit)

Correct, this was meant to be hidden on tablet down. I will improve this further once we have backend editing function of those api key permissions.

> (Very) slight difference in cancel button background colour (keyname / add)

Transparency style added here.

> Very thin menu in tablet resolution

Good catch, this real estate utilization is poor in the tablet range. I've applied a fix here, please have another look.

I also changed 'map' -> 'object' field type terminology (Henri's request). And a bunch of other smaller details that I missed on the first round. Please take another look and let me know how you go.

**Comment by fonty1** [comment]
> Looks a lot better now! I noticed that adding API keys for stream does not work when creating a new stream. After saving it works as expected. Maybe not part of this PR but we probably should disable the add button when stream does not yet have id.

Interesting, good catch. I'll see what can be done here.

**Comment by juhah** [comment]
Looks very good, all issues fixed except the minor background difference in the cancel button.

Screen Shot 2019-03-18 at 13 15 34

In profile page (tablet resolution), the add key form background extends to the edge which looks a bit awkward:
Screen Shot 2019-03-18 at 13 14 25

**Comment by fonty1** [comment]
> Looks very good, all issues fixed except the minor background difference in the cancel button.

> In profile page (tablet resolution), the add key form background extends to the edge which looks a bit awkward:

Those should be fixed - assuming you mean the hover state for the cancel button? That cancel button does follow the style guide but the style guide isn't prepared for this edge case of a light grey container. I've removed it for now which looks a bit better.

File: app/src/userpages/components/ProfilePage/IntegrationKeyHandler/index.jsx [hunk 1]
```diff
@@ -52,9 +50,14 @@ export class IntegrationKeyHandler extends Component<Props> {
                     createWithValue
                 />
                 <AddKeyField
-                    label={I18n.t('userpages.profilePage.ethereumAddress.addNewAddress')}
+                    // $FlowFixMe
+                    label={this.props.integrationKeys[0] ?
```
[the remark this answers is not in the document]
Comment (fonty1) [reply app/src/userpages/components/ProfilePage/IntegrationKeyHandler/index.jsx]: thanks @timoxley, good catch and I appreciate the code snippet 👏 👌

I will make sure I circle back to these sorts of fixes in the future before merge.

**Pull request merged**